### ISSUE TITLE
ValueError: month must be in 1..12

### ISSUE THREAD

**Issue by JulienPalard** [aroberge/friendly#229]
L'exception du jour:

```
$ python -m friendly

Friendly Console version 0.3.63. [Python version: 3.9.4]

>>> from datetime import date
>>> date(2021, 13, 1)

Traceback (most recent call last):                                                                                                                             
  File "<friendly-console:2>", line 1, in <module>                                                                                                             
    date(2021, 13, 1)                                                                                                                                          
ValueError: month must be in 1..12                                                                                                                             
>>> why()

No information is known about this exception. Please report this example to https://github.com/aroberge/friendly/issues                      
```

On my formatter I also got:

```
Une exception `ValueError` indique qu'une fonction ou une opération
a reçu un argument du bon type, mais une valeur inappropriée.
```

which is already nice, maybe it could be enhanced by interweaving the function name? 

```
La fonction `date` a reçu un argument du bon type mais une valeur inappropriée. 
```

**Comment by aroberge** [comment]
Yes, I've barely started on ValueError cases from the standard library. I want to take care of a few more SyntaxError cases, and then I will start tackling these.

What you got on your formatter is what I called the generic case (what does an error of type X mean); in interactive use, this is the answer to `what()`.  The more specific information is obtained by doing a more in depth inspection, based on a case by case analysis; this is the result of `why()` in the interactive terminal.  I suppose that, for ValueError cases, I could have a "generic specific" information along the lines of what you suggest, when I have no other information to give.  I could possibly even provide the docstring which, in this case, would be:
```python
>>> date.__doc__
'date(year, month, day) --> date object'
```
To summarize: when I do not recognize the error message for ValueError cases, I could 1) give the function name and 2) give the function's docstring (unfortunately not translated).

**Comment by aroberge** [comment]
Work in progress:
```
> python -m friendly

Friendly Console version 0.3.63. [Python version: 3.8.4]

>>> from datetime import date
>>> d = date(2021, 13, 1)

Traceback (most recent call last):
  File "<friendly-console:2>", line 1, in <module>
    d = date(2021, 13, 1)
ValueError: month must be in 1..12
>>> why()

I do not recognize this error message. I am guessing that the problem is with the
function `date`. Its docstring is:

date(year, month, day) --> date object
>>> set_lang('fr')
>>> why()

Je ne reconnais pas ce message d’erreur. Je devine que le problème est avec la
fonction `date`. Sa docstring est :

date(year, month, day) --> date object
>>>
```

**Comment by aroberge** [comment]
```
> python -m friendly --lang fr

Friendly Console version 0.3.63. [Python version: 3.8.4]

>>> from datetime import date
>>> d = date(2021, 13, 1)

Traceback (most recent call last):
  File "<friendly-console:2>", line 1, in <module>
    d = date(2021, 13, 1)
ValueError: month must be in 1..12

Avez-vous spécifié un mois inexistant ?
>>> why()

Je crois que vous avez indiqué une valeur non valide pour le mois dans un objet de
type date. Les valeurs valides sont les entiers de 1 à 12.
>>>
```
will be part of the next release.

**Comment by aroberge** [comment]
New release available.

Closing this for now. If you think that a better explanation could be provided, feel free to reopen this issue.
